bin/utils.js

```
@@ -0,0 +1,123 @@
+var shell = require('shelljs');
+
+var fs = require('fs');
+var path = require('path');
+
+// paths
+var cwd = process.cwd();
+var binDir = __dirname;
+var rootDir = path.resolve(binDir, '../');
+var web3Dir = path.join(rootDir, '/node_modules/@solana/web3.js');
+var bpfSdkDir = path.join(web3Dir, '/bpf-sdk');
+
+// shells
+var sdkInstaller = path.join(binDir, '/bpf-sdk-install.sh');
+var programBuilder = path.join(bpfSdkDir, 'rust/build.sh');
+
+function exec(cmd, fn) {
+  shell.exec(cmd, function (code, stdout) {
+    if (code === 0) {
+      fn(stdout)
+    } else {
+      console.log('Some error occurs, exit.')
+    }
+  });
+}
+
+// The Cargo.tomal package name may not equals the program name,
+// so we so we need to compatible this.
+function getSoFilePath(program, profilePath) {
+  try {
+    // If the most likely file exists, just return it.
+    var soFilePath = path.join(profilePath, (program + '').replace(/\-/g, '_') + '.so');
+    if (fs.existsSync(soFilePath)) {
+      return soFilePath;
+    } else {
+      // Find the first .so file in the directory
+      var files = fs.readdirSync(profilePath);
+      var soFile = '';
+      files.every(function(file) {
+        return !(/.*\.so$/.test(file) && (soFile = file));
+      });
+      return path.join(profilePath, soFile);
+    }
+  } catch(e) {
+    return '';
```

does it make sense to return '' here? maybe just let error propagate?